v1/brokers/aws_sqs.go

```diff
@@ -206,13 +206,14 @@ func (b *AWSSQSBroker) consumeOne(delivery *sqs.ReceiveMessageOutput, taskProces
 		return fmt.Errorf("task %s is not registered", sig.Name)
 	}
 
-	// Delete message after consuming successfully
-	err := b.deleteOne(delivery)
+	err := taskProcessor.Process(sig)
 	if err != nil {
```

Comment:
Isn't it a bug? It would only delete the task if task processor failed.

Reply:
Oh, it does appear that way! I believe this should be `if err == nil`!

Reply:
Not sure why the diff is misleading.

Code was changed here: https://github.com/RichardKnop/machinery/commit/74f2b63fd3304bfaa55e39a3b285b5b839394c57

And code in the master branch looks fine: https://github.com/RichardKnop/machinery/blob/master/v1/brokers/sqs/sqs.go#L197

Reply:
I think this has been corrected in master branch already. Please check the code in the master branch if it is correct. I am not too familiar with the SQS broker code as I haven't written it (I know much more about AMQP and Redis which was originally written by me).